File: review-rating-service/src/main/java/com/mentorhub/reviewratingservice/model/Review.java
package com.mentorhub.reviewratingservice.model;
import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;
import java.util.UUID;

@Entity @Table(name = "reviews") @Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Review {
    @Id @Column(name = "\"Id\"") private UUID id;
    @Column(name = "\"StudentProfileId\"") private UUID studentProfileId;
    @Column(name = "\"TutorProfileId\"") private UUID tutorProfileId;
    @Column(name = "\"BookingId\"") private UUID bookingId;
    @Column(name = "\"BatchEnrollmentId\"") private UUID batchEnrollmentId;
    @Column(name = "\"ReviewType\"") private int reviewType;
    @Column(name = "\"Rating\"") private int rating;
    @Column(name = "\"Comment\"", columnDefinition = "text") private String comment;
    @Column(name = "\"Sentiment\"", length = 20) private String sentiment;
    @Column(name = "\"SentimentConfidence\"") private Double sentimentConfidence;
    @Column(name = "\"ReviewWindowExpiresAtUtc\"") private Instant reviewWindowExpiresAtUtc;
    @Column(name = "\"CreatedAtUtc\"") private Instant createdAtUtc;
}

File: UserManagment/UserManagment.Domain/Entities/Review.cs
using UserManagment.Domain.Enums;

namespace UserManagment.Domain.Entities;

public class Review
{
    public Guid Id { get; set; }
    public Guid StudentProfileId { get; set; }
    public Guid TutorProfileId { get; set; }
    public Guid? BookingId { get; set; }
    public Guid? BatchEnrollmentId { get; set; }
    public ReviewType ReviewType { get; set; }

    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
    public string? Sentiment { get; set; }
    public decimal? SentimentConfidence { get; set; }
    public DateTime? ReviewWindowExpiresAtUtc { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public StudentProfile StudentProfile { get; set; } = null!;
    public TutorProfile TutorProfile { get; set; } = null!;
    public BatchEnrollment? BatchEnrollment { get; set; }
}
